### ISSUE TITLE
Implement Mappable in an extension

### ISSUE THREAD

**Issue by EliaCereda** [Hearst-DD/ObjectMapper#264]
I'd like to use ObjectMapper with some classes that I do not control, so I'd have to implement the `Mappable` protocol from an extension, which is currently impossible because of the `required init?` method.

If I try to implement `init?(map: Map)` from an extension the compiler gives me this error:
```
initializer requirement 'init' can only be satisfied by a `required` initializer in the definition of non-final class 'ClassName'
```

How could I make it work? I saw that ObjectMapper used to have a `newInstance(map: Map)` method. I'd have been able to use that, since it doesn't involve the restrictions on `required init()`s.

**Comment by tosbaha** [comment]
Try to mark your class as ```final``` and use  ```convenience init?``` in the extension.

**Comment by EliaCereda** [comment]
Except those aren't _my_ classes so I can't change them at all. I've edited ObjectMapper to use the old `newInstance(map: Map)` method, but it still doesn't work like I'd want it to.

Using `newInstance` I'm able to add Mappable conformance to the classes I need, but since Swift doesn't allow me to override methods in an extension I can't implement Mappable in class clusters.

**Comment by tosbaha** [comment]
I also experienced same issue with other serializers like [Cereal](https://github.com/Weebly/Cereal) That is why I stopped using them. I currently use some kind of proxy class to get data from original class to proxy class and then use  [swift-serializer](https://github.com/Mailcloud/swift-serializer) to turn object to JSON or dictionary. It doesn't support structs or have all advanced properties of mentioned libraries but it works for me.

**Comment by valeriomazzeo** [comment]
I think your issue can probably relate to #260

**Comment by antrix1989** [comment]
+1.

**Comment by camina-apps** [comment]
+1

**Comment by sbhklr** [comment]
+1

**Comment by palusik86** [comment]
+1 and current protocol Mappable creates some issue with Realm.

**Comment by tristanhimmelman** [comment]
This would be a nice addition but I see no simple way to get rid of the `init?` function. I am marking this as an `enhancement` to keep track of the idea

**Comment by tristanhimmelman** [comment]
I have just released v1.5.0 which now supports implementing ObjectMapper within an extension. If you wish to do this, please use the `StaticMappable` protocol. With this update `StaticMappable` no longer requires the `init?(_ map: Map)` function which was causing difficulties.

Hope this helps you all
